modules/common/src/main/scala/surge/internal/akka/ActorWithTracing.scala

```diff
@@ -3,29 +3,75 @@
 package surge.internal.akka
 
 import akka.AroundReceiveActor
-import io.opentracing.Scope
-import surge.internal.utils.SpanSupport
-import surge.tracing.TracedMessage
+import akka.actor.ActorRef
+import io.opentracing._
+import surge.tracing.{ SpanExtensions, TracePropagation, TracedMessage }
 
-trait ActorWithTracing extends AroundReceiveActor with SpanSupport {
-  private var activeScope: Option[Scope] = None
+trait ActorWithTracing extends AroundReceiveActor with ActorOps with SpanExtensions {
```

Comment:
One change is that every message gets a span called "ActorName:Message" (i.e. "PersistentActor:ProcessCommand[AddItem]" )

When the message is a TraceableMessage the span gets a parent span. Otherwise, the span doesn't have a parent. 

Can this lead to too many spans ? Yes - but we can use the sampler configuration to discard spans we don't need.